packages/hadron-document/src/document.ts

```diff
@@ -109,40 +109,40 @@ export class Document extends EventEmitter {
    * where the update only succeeds when the changed document's elements have
    * not been changed in the background.
    *
+   * `query` and `updateDoc` may use $getField and $setField if field names
+   * contain either `.` or start with `$`. These operators are only available
+   * on MongoDB 5.0+. (Note that field names starting with `$` are also only
+   * allowed in MongoDB 5.0+.)
+   *
    * @param {Object} alwaysIncludeKeys - An object whose keys are used as keys
-   *     that are always included in the generated query.
+   *     that are always included in the generated query. Dots inside key names
+   *     are interpreted as referring to nested properties.
    *
    * @returns {Object} An object containing the `query` and `updateDoc` to be
    * used in an update operation.
    */
   generateUpdateUnlessChangedInBackgroundQuery(
-    alwaysIncludeKeys: BSONObject | null = null
+    alwaysIncludeKeys: string[] = []
   ): {
     query: BSONObject;
-    updateDoc: { $set?: BSONObject; $unset?: BSONObject };
+    updateDoc: { $set?: BSONObject; $unset?: BSONObject } | BSONArray;
```

Comment:
Totally can see that this PR is marked as a breaking change with a `!`, but still wanted to highlight that cloud seems to have [very hard expectations](https://github.com/10gen/mms/blob/4513c476627316e06e94b2efc26b1e1aec182529/client/js/project/metrics/components/DataExplorerFindResults.js#L107-L110) for this to return set / unset object here. Just wondering if we checked with them whether they would be able to handle this new format with their backed at all, maybe we should create a downstream ticket for them to look into this 🤔